core/write-file.js

```diff
@@ -0,0 +1,22 @@
+/*
+  Copyright 2015 Google Inc. All Rights Reserved.
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+var fs = require('fs');
+var mkdirp = require('mkdirp');
+var dirname = require('path').dirname;
+
+// Like `fs.writeFileSync`, but creates parent directories as needed.
+module.exports.writeFileSync = function (path, contents) {
```

Comment:
Not sure it's worth having an entire new file for this, can we just do it directly in writeStringFile?